File: Solutions/0728- Self Dividing Numbers/Solution.java
class Solution {
    public List<Integer> selfDividingNumbers(int left, int right) {
        List<Integer> answer = new ArrayList<>();

        for (int i = left; i <= right; i++) {
            int number = i;
            boolean selfDiving = true;
            
            while (number > 0) {
                int remainder = number % 10;
                if (remainder == 0 || i % remainder != 0) {
                    selfDiving = false;
                    break;
                }
                number /= 10;
            }
            if (selfDiving) {
                answer.add(i);
            }
        }
        
        return answer;

    }

}

File: Solutions/0728- Self Dividing Numbers/Solution.py
class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        answer = list()

        for i in range(left,right+1):
            number = i
            selfDiving = True

            while number > 0:
                remainder = number % 10
                if remainder == 0 or i % remainder != 0:
                    selfDiving = False
                    break
                number //= 10
            
            if selfDiving:
                answer.append(i)

        return answer
